app/core/Permissions/index.ts

```diff
@@ -100,24 +94,19 @@ export const addPermittedAccounts = (
     }
   }
 
-  const selectedAccounts: SelectedAccount[] = addresses.map(
-    (address, index) => ({ address, lastUsed: Date.now() - index }),
-  );
-
-  const newSortedAccounts = orderBy<SelectedAccount>(
-    [...existing.value, ...selectedAccounts],
-    'lastUsed',
-    'desc',
-  );
+  const newPermittedAccounts = [
+    ...addresses,
+    ...existingPermittedAccountAddresses,
+  ];
```

Comment:
Recommend defensively deduping this (assumes addresses are cased the same, which for us should mean lowercased):
```suggestion
  const newPermittedAccounts = Array.from(new Set([
    ...addresses,
    ...existingPermittedAccountAddresses,
  ]));
```